cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/sync/class-push-sync.php

```diff
@@ -163,6 +175,54 @@ public function rest_get_queue_status() {
 		);
 	}
 
+	/**
+	 * Send an explicit request for large videos so that transformations can be applied async.
+	 *
+	 * @param \WP_REST_Request $request
+	 *
+	 * @return \WP_REST_Response
+	 */
+	public function rest_video_explicit_upload( \WP_REST_Request $request ) {
+		$req_body = json_decode( $request->get_body(), true );
+
+		if ( ! $this->plugin->components['sync']->is_pending( $req_body['attachment_id'] ) ) {
+			$res = $this->media->video->queue_eager_video( $req_body );
+			return rest_ensure_response( ! is_wp_error( $res ) ? 'ok' : 'error' );
+		}
+
+		return rest_ensure_response( 'pending' );
+	}
+
+	/**
+	 * Wait for Cloudinary's response on the eager sync status.
+	 *
+	 * @param \WP_REST_Request $request
+	 *
+	 * @return \WP_REST_Response
```

Comment:
@dugajean 

I think we are missing the actual returns for this callback.
Let's add something that eventually Cloudinary can track.